src/assign.cpp

```diff
@@ -257,7 +279,40 @@ bool fba::Assignment::petal_count_max(
 }
 
 
```

Comment:
This basically exactly mirrors a pair of per-petal functions doing the same thing.